manifests/init.pp

```diff
@@ -420,6 +424,11 @@
     validate_string($ssl_protocol)
   }
 
+  if $yum_max_speed {
+    validate_integer($yum_max_speed)
+    to_bytes($yum_max_speed)
```

Comment:
That line looks okay, but we would expect users to supply something like "4 Kb", "60 Mb" or "1 Gb" so there is no need to `validate_integer` in the line above, could use `validate_string` if you would like.